inst/stan/functions/gaussian_process.stan

```diff
@@ -57,15 +79,24 @@ vector update_gp(matrix PHI, int M, real L, real alpha,
     for(m in 1:M){
       diagSPD[m] =  sqrt(spd_se(alpha, unit_rho, sqrt(lambda(L, m))));
     }
-  }else if (type == 1) {
+  } else if (type == 1) {
+    for(m in 1:M){
+      diagSPD[m] =  sqrt(spd_ou(alpha, unit_rho, sqrt(lambda(L, m))));
+    }
+  } else if (type == 2) {
+    for(m in 1:M){
+      diagSPD[m] =  sqrt(spd_matern32(alpha, unit_rho, sqrt(lambda(L, m))));
+    }
+  } else if (type == 3) {
     for(m in 1:M){
-      diagSPD[m] =  sqrt(spd_matern(alpha, unit_rho, sqrt(lambda(L, m))));
+      diagSPD[m] =  sqrt(spd_matern52(alpha, unit_rho, sqrt(lambda(L, m))));
     }
   }
   SPD_eta = diagSPD .* eta;
```

Comment:
This suggests we can construct diagSPD directly: https://avehtari.github.io/casestudies/Motorcycle/motorcycle_gpcourse.html#4_Heteroskedastic_GP_with_Hilbert_basis_functions

Reply:
I'm going to look at this in another PR.